Composite/Plugins/Functions/FunctionProviders/StandardFunctionProvider/Pages/GetPageIdFunction.cs

```diff
@@ -27,28 +28,74 @@ public override object Execute(ParameterList parameters, FunctionContextContaine
                 SitemapScope = SitemapScope.Current;
             }
 
-            Guid pageId = Guid.Empty;
+            var pageId = GetCurrentPageId();
 
             switch (SitemapScope)
             {
                 case SitemapScope.Current:
-                    pageId = PageRenderer.CurrentPageId;
-                    break;
+                    return pageId;
                 case SitemapScope.Parent:
                 case SitemapScope.Level1:
                 case SitemapScope.Level2:
                 case SitemapScope.Level3:
                 case SitemapScope.Level4:
-                    IEnumerable<Guid> pageIds = PageStructureInfo.GetAssociatedPageIds(PageRenderer.CurrentPageId, SitemapScope);
-                    pageId = pageIds.FirstOrDefault();
-                    break;
+                    var pageIds = PageStructureInfo.GetAssociatedPageIds(pageId, SitemapScope);
+                    return pageIds.FirstOrDefault();
                 default:
                     throw new NotImplementedException("Unhandled SitemapScope type: " + SitemapScope.ToString());
             }
+        }
+
+        private Guid GetCurrentPageId()

```

Comment:
minor bug
wWen adding new metadata to the page then pageId is empty.
We can merge without fixing this.

Reply:
should work fine for this case

Reply:
Did you validate? For some reason, I still have it default.
![image](https://user-images.githubusercontent.com/14817424/71406154-06ad3b00-2640-11ea-9ed2-c96cadc20ded.png)

Reply:
removed this widget from list, because it is for internal use only